src/parsed.rs

```diff
@@ -242,3 +243,91 @@ impl<'de> ListVisitor<'de> for List {
         Ok(self)
     }
 }
+
+/// A structured-field type, supporting parsing and serialization.
+pub trait FieldType: Sealed {
```

Comment:
I don't especially like the name FieldType

How about `StructuredField` or `StructuredFieldValue` which is the concept they're trying to represent.
Having `Type` in the name seems redundant.